src/compiler/program.ts

```diff
@@ -1714,186 +1714,172 @@ namespace ts {
-                            if (nodes === (<ParameterDeclaration>parent).modifiers) {
-                                diagnostics.push(createDiagnosticForNodeArray(nodes, Diagnostics.parameter_modifiers_can_only_be_used_in_a_ts_file));
-                                return;
-                            }
-                            break;
-                        case SyntaxKind.CallExpression:
-                        case SyntaxKind.NewExpression:
-                        case SyntaxKind.ExpressionWithTypeArguments:
-                        case SyntaxKind.JsxSelfClosingElement:
-                        case SyntaxKind.JsxOpeningElement:
-                            // Check type arguments
-                            if (nodes === (<CallExpression | NewExpression | ExpressionWithTypeArguments | JsxOpeningLikeElement>parent).typeArguments) {
-                                diagnostics.push(createDiagnosticForNodeArray(nodes, Diagnostics.type_arguments_can_only_be_used_in_a_ts_file));
-                                return;
-                            }
-                            break;
-                    }
+                            return;
+                        }
+                        break;
+                    case SyntaxKind.Parameter:
+                        // Check modifiers of parameter declaration
+                        if (nodes === (<ParameterDeclaration>parent).modifiers) {
+                            diagnostics.push(createDiagnosticForNodeArray(nodes, Diagnostics.parameter_modifiers_can_only_be_used_in_a_ts_file));
+                            return;
+                        }
+                        break;
+                    case SyntaxKind.CallExpression:
+                    case SyntaxKind.NewExpression:
+                    case SyntaxKind.ExpressionWithTypeArguments:
+                    case SyntaxKind.JsxSelfClosingElement:
+                    case SyntaxKind.JsxOpeningElement:
+                        // Check type arguments
+                        if (nodes === (<CallExpression | NewExpression | ExpressionWithTypeArguments | JsxOpeningLikeElement>parent).typeArguments) {
+                            diagnostics.push(createDiagnosticForNodeArray(nodes, Diagnostics.type_arguments_can_only_be_used_in_a_ts_file));
+                            return;
+                        }
+                        break;
+                }
 
-                    for (const node of nodes) {
-                        walk(node);
-                    }
+                for (const node of nodes) {
+                    walk(node);
                 }
+            }
 
-                function checkModifiers(modifiers: NodeArray<Modifier>, isConstValid: boolean) {
-                    for (const modifier of modifiers) {
-                        switch (modifier.kind) {
-                            case SyntaxKind.ConstKeyword:
-                                if (isConstValid) {

```

Comment:
where is this checked if not here? I assume we're supposed to get an error for `const` on method declarations in both Javascript and Typescript.

Reply:
Why is `const` special?
As I see it this function reports Typescript syntax that is not valid in JS files and there's nothing TS specific about `const`.

What you really want is all grammar errors in TS and JS files. Hence I opened #29213 to propose exactly that.

Reply:
So the const modifier check is otherwise handled in the checker's grammar errors? 

I read #29213, and it makes sense, although syntactic errors block semantic errors (or did at one time), so we'd need to test that the editor experience doesn't get worse.

However, removing this check here would then remove the error in non-ts-tcheck JS files until #29213 is fixed, right? If that's the case we should wait until that is done before making this change.